src/compressed_tensors/quantization/utils/helpers.py

```diff
@@ -73,14 +83,46 @@ def calculate_qparams(
 
     bit_min, bit_max = calculate_range(quantization_args, device)
     bit_range = bit_max - bit_min
-    zp_dtype = quantization_args.pytorch_dtype()
+
+    if (
+        quantization_args.num_bits == 4
+        and quantization_args.type == QuantizationType.FLOAT
+    ):
```

Comment:
Do we ever overload functions? Given the number of times we do this check in this code, i think it might be more readable if we have a `calculate_qparams` for FLOAT and another for INT, rather than passing both cases into a single function with several if/else checks on the same conditions.

Reply:
I think it probably makes sense to just move the redundant code in a helper/utils that we call at the moment.

Otherwise, this change is very specific to nvfp4, so I dont think we need to yet optimize separations between float/int (e.g if we decide to support mxfp4, the scale processing may be different).